packages/odyssey-react-mui/src/labs/Drawer.tsx

```diff
@@ -0,0 +1,224 @@
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Drawer as MuiDrawer } from "@mui/material";
+
+import {
+  useOdysseyDesignTokens,
+  DesignTokens,
+} from "../OdysseyDesignTokensContext";
+
+import { Button } from "../Button";
+import { Box } from "../Box";
+import { CloseIcon } from "../icons.generated";
+import {
+  memo,
+  ReactNode,
+  useState,
+  useEffect,
+  useRef,
+  ReactElement,
+} from "react";
+
+import styled from "@emotion/styled";
+import { useTranslation } from "react-i18next";
+import type { AllowedProps } from "../AllowedProps";
+
+export const variantValues = ["temporary", "persistent"] as const;
+
+export type DrawerProps = {
+  /**
+   * An optional Button object to be situated in the Drawerfooter. Should almost always be of variant `primary`.
+   */
+  callToActionFirstComponent?: ReactElement<typeof Button>;
+  /**
+   * An optional Button object to be situated in the Drawer footer, alongside the `callToActionPrimaryComponent`.
+   */
+  callToActionSecondComponent?: ReactElement<typeof Button>;
+  /**
+   * An optional Button object to be situated in the Drawer footer, alongside the other two `callToAction` components.
+   */
+  callToActionLastComponent?: ReactElement<typeof Button>;
+  /**
+   * The content of the Drawer. May be a `string` or any other `ReactNode` or array of `ReactNode`s.
+   */
+  children?: ReactNode;
+  /**
+   * When set to `true`, the Drawer will be visible.
+   */
+  isOpen?: boolean;
+  /**
+   * Callback that controls what happens when the Drawer is dismissed
+   */
+  onClose: () => void;
+  /**
+   * The title of the Drawer
+   */
+  title?: string;
+  /**
+   * Type of Drawer
+   */
+  variant?: (typeof variantValues)[number];
+  ariaLabel: string;
+} & AllowedProps;
+
+const DrawerHeader = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>`
+  position: sticky;
+  display: flex;
+  top: 0;
+  background-color: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.HueNeutralWhite};
+  justify-content: space-between;
+  align-items: center;
+  margin: 0;
+  padding: ${({ odysseyDesignTokens }) => odysseyDesignTokens.Spacing4} 0;
+  font-family: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.TypographyFamilyHeading};
+  color: ${({ odysseyDesignTokens }) => odysseyDesignTokens.HueNeutral900};
+
+  h2 {
+    margin: 0;
+    font-size: ${({ odysseyDesignTokens }) =>
+      odysseyDesignTokens.TypographySizeHeading5};
+    font-weight: 500;
+  }
+`;
+
+const DrawerFooter = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>`
+  display: flex;
+  position: sticky;
+  bottom: 0;
+  background-color: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.HueNeutralWhite};
+  flex-direction: row;
```

Comment:
This is the default value, probably don't need this line. Also, same nit as above, can we move flex stuff together?